torch/csrc/jit/frontend/concrete_module_type.cpp

```diff
@@ -269,9 +297,10 @@ void ConcreteModuleTypeBuilder::addBuiltinFunction(
 
 void ConcreteModuleTypeBuilder::addModule(
     std::string name,
-    std::shared_ptr<ConcreteModuleType> meta) {
-  modules_.emplace_back(
-      ConcreteModuleTypeBuilder::ModuleInfo{std::move(name), std::move(meta)});
+    std::shared_ptr<ConcreteModuleType> meta,
+    TypePtr containedTypeHint) {
```

Comment:
when would we be adding `containedTypeHint` only to one specific module ?

Reply:
```
class Mod(torch.nn.Module):
            __annotations__ = {"d": Dict[str, ModuleInterface]}

            def __init__(self):
                super().__init__()
                self.d = torch.nn.ModuleDict({"module": ImplementsInterface()})

            def forward(self, x: torch.Tensor, key: str) -> Any:
                return self.d[key].forward(x)
```

The way I've been thinking about it is that when building the concrete type for `Mod`, we add `Dict[str, ModuleInterface]` as a type hint to the `ConcreteModuleType` for `Mod` *for its attribute named `d`*. I haven't been thinking about the hint as something that applies to the module referred to by `d` in all circumstances.

When/if we eventually fix module aliasing, the module referred to by `d` could appear in multiple places, so setting the hint directly on its `ConcreteModuleType` wouldn't make sense.